### PR TITLE
Initial charging and discharging of avatar energy

### PR DESCRIPTION
**Pull request by jagwire**
Initial charging and discharging of avatar energy based on movement and audio.

- exposed energy property to MyAvatar script interface
- energy recalculated each update step
- energyBar.js draws overlay and details usage of MyAvatar energy property

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by AndrewMeadows** [comment]
ok to test

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -283,6 +283,18 @@ void MyAvatar::update(float deltaTime) {
     head->setAudioLoudness(audio->getLastInputLoudness());
     head->setAudioAverageLoudness(audio->getAudioAverageInputLoudness());
 
+    
+    currentEnergy += energyChargeRate;
+    currentEnergy -= getAccelerationEnergy();
+    
+    if(didTeleport()) {
+        currentEnergy = 0;
+    }
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: The coding standard sections 3.5.3 and 3.5.4 require that the literal value here be an explicit float:  0.0f.

https://docs.highfidelity.com/docs/coding-standard#section-3-5-3-floating-point-constants-should-always-be-written-with-a-digit-before-the-decimal-point-

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -396,6 +397,19 @@ public slots:
 
     AtRestDetector _hmdAtRestDetector;
     bool _lastIsMoving { false };
+    
+    float AVATAR_MOVEMENT_ENERGY_CONSTANT { 0.001 };
+    float AUDIO_ENERGY_CONSTANT { 0.000001f };
+    float MAX_AVATAR_MOVEMENT_PER_FRAME = 30.0f;
+    float currentEnergy { 0.0f };
+    float energyChargeRate { 0.003 };
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.h]: Coding standard section 3.5.4 requires a trailing 'f':

https://docs.highfidelity.com/docs/coding-standard#section-3-5-4-when-using-a-single-precision-float-type-include-the-trailing-f-

File: interface/src/avatar/MyAvatar.h [hunk 3]
```diff
@@ -396,6 +397,19 @@ public slots:
 
     AtRestDetector _hmdAtRestDetector;
     bool _lastIsMoving { false };
+    
+    float AVATAR_MOVEMENT_ENERGY_CONSTANT { 0.001 };
+    float AUDIO_ENERGY_CONSTANT { 0.000001f };
+    float MAX_AVATAR_MOVEMENT_PER_FRAME = 30.0f;
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.h]: coding standard use `{ 30.0f }` format

**Comment by hifi-gustavo** [comment]
Build console output for commit: b679d012b51a196abaa4b97b9988f6292145608a available at: https://gist.github.com/a4c9b925ccad31bda5ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6852.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6852.dmg

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Pull request closed without merging**